typescript/sdk/src/crud/HookModule.ts

```diff
@@ -0,0 +1,52 @@
+import { Address, ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+
+import { HookFactories } from '../hook/contracts.js';
+import { EvmHookReader, HookReader } from '../hook/read.js';
+import { HookConfig } from '../hook/types.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import {
+  EthersV5Transaction,
+  SupportedProtocolType,
+} from '../providers/ProviderType.js';
+
+import { CrudModule, CrudModuleArgs } from './AbstractCrudModule.js';
+
+export abstract class HookModule<
+  TProtocol extends SupportedProtocolType,
+> extends CrudModule<TProtocol, HookConfig, HookFactories> {
+  protected abstract reader: HookReader;
+
+  public async read(address: Address): Promise<HookConfig> {
+    return await this.reader.deriveHookConfig(address);
+  }
+}
+
+// WIP example implementation of EvmHookModule
+export class EvmHookModule extends HookModule<ProtocolType.Ethereum> {
+  protected logger = rootLogger.child({ module: 'EvmHookModule' });
+  protected reader: EvmHookReader;
+
+  protected constructor(
+    multiProvider: MultiProvider,
+    args: Omit<
+      CrudModuleArgs<ProtocolType.Ethereum, HookConfig, HookFactories>,
+      'provider'
+    >,
+  ) {
+    super({
+      ...args,
+      provider: multiProvider.getProvider(args.chain),
```

Comment:
Why pass up a specific provider instead of the whole MP as we typically do?

Reply:
we have the protocoltype->providertype mapping to be able to set expected types in the top level abstract module, and this is part of that enforcement

I think it would be odd if we had a multiprovider here but the specific returned tx type in update() is a specific EthersV5Transaction

We can't use the multiprovider at the top-level because it is EVM specific